MapboxDirections/MBBoundingBox.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+import CoreLocation
+
+/**
+ A bounding box represents a geographic region.
+ */
+@objc(MBBoundingBox)
+public class BoundingBox: NSObject, Codable {
```

Comment:
The map SDK has a similar struct named [`MGLCoordinateBounds`](https://github.com/mapbox/mapbox-gl-native/blob/f50d8dbe3afa2556b1ef404e39ef897e0093cd0c/platform/darwin/src/MGLGeometry.h#L69-L75), and MapboxGeocoder.swift has a CLRegion subclass named [RectangularRegion](https://github.com/mapbox/MapboxGeocoder.swift/blob/0af564695d0153bbfddeae729bac8ae22e89334d/MapboxGeocoder/MBRectangularRegion.swift). It would be great if we could make this class a struct, for value semantics; otherwise, a subclass of CLRegion would at least make some of the capabilities more discoverable.

Reply:
Hm, yeah, it's unfortunate that MapboxDirections doesn't depend on those. Would be nice to share a core set of data types, similar to what gl-native is doing.

I went back and forth with CLRegion but didn't find it adding any values CoordinateBounds makes sense for consistency. How about `{MB}/CoordinateBounds`?

Reply:
Turf has a [BoundingBox](https://github.com/mapbox/turf-swift/blob/master/Sources/Turf/BoundingBox.swift) but we are partially striving for turf-js terminology there.

Reply:
GeoJSON terminology makes more sense for a library that’s very much about manipulating GeoJSON, such as Turf, but unless we have this library depend on Turf – not necessarily a bad idea – I kind of like CoordinateBounds for consistency with the map SDK. I agree that CLRegion isn’t adding much in this case; MapboxGeocoder.swift has that to be more similar to MapKit’s and Core Location’s geocoders.

> Swift struct wouldn't bridge to Obj-C.

A C struct does bridge to Swift, so implement it in an Objective-C header.